src/main/java/org/broadinstitute/hellbender/utils/reference/ReferenceUtils.java

```diff
@@ -107,4 +116,40 @@ public static byte[] getRefBaseAtPosition(final ReferenceSequenceFile reference,
     public static byte[] getRefBasesAtPosition(final ReferenceSequenceFile reference, final String contig, final int start, final int length) {
         return reference.getSubsequenceAt(contig, start, start+length-1).getBases();
     }
+
+    /**
+     * Given a reference data source and a sequence interval, calculates the MD5 for the given sequence.
+     *
+     * Note: does not close the ReferenceDataSource it's passed.
+     * Note: MD5 calculation mimics the MD5 calculation in {@link picard.sam.CreateSequenceDictionary}: allows IUPAC bases
+     *       but uppercases all bases.
+     *
+     * @param referencePath The path to the reference.
+     * @param interval The interval of the sequence.
+     * @return the sequence's MD5 as a String.
+     */
+    public final static String calculateMD5(GATKPath referencePath, SimpleInterval interval){
+        MessageDigest md5;
+        try {
+            md5 = MessageDigest.getInstance("MD5");
+        }
+        catch(NoSuchAlgorithmException exception){
+            throw new GATKException("Incorrect MessageDigest algorithm specified in calculateMD5()", exception);
+        }
+
+        try(final ReferenceDataSource source = ReferenceDataSource.of(referencePath.toPath(), true)) {
```

Comment:
Add comment: we pass in "true" as the second arg to `ReferenceDataSource.of()` to tell it not to modify IUPAC bases or capitalization for us.